File: thi/src/service/ProductService.js
import axios from "axios";

const getAll = async () => {
    try {
        const result = await axios.get(`http://localhost:8080/productList`)
        return result.data
    }catch (error){
        return error;
    }
}

const createProduct = async (products) => {
    try {
        const result = await axios.post(`http://localhost:8080/productList`,products)
        return result.data
    }catch (error){
        return error;
    }
}
const UpdateProduct = async  (id, products) => {
    try {
         const result = await axios.put(`http://localhost:8080/productList/${id}`,products)
        return result.data
    }catch (error){
        return error;
    }
}
const findById = async (id) => {
    try {
        const result = await axios.get(`http://localhost:8080/productList/${id}`)
        return result.data
    }catch (error){
        return error;
    }
}
const deleteProduct = async (id) => {
    try {
        const result = await axios.delete(`http://localhost:8080/productList/${id}`)
        return result.data
    }catch (error){
        return error;
    }
}
const searchByName = async (name) => {
    try {
        const result = await axios.get(`http://localhost:8080/productList?name_like=${name}`)
        return result.data;
    }catch (error){
        return error;
    }
}
const productService = {
    getAll,
    createProduct,
    UpdateProduct,
    findById,
    deleteProduct,
    searchByName
}
export default productService;

File: ConnectDBReact/src/main/java/com/example/connectdbreact/service/impl/ProductService.java
package com.example.connectdbreact.service.impl;

import com.example.connectdbreact.model.Product;
import com.example.connectdbreact.repository.IProductRepository;
import com.example.connectdbreact.service.IProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository iProductRepository;

    @Override
    public Page<Product> findAll(Pageable pageable) {
        return iProductRepository.getAll(pageable);
    }

    @Override
    public void create(Product product) {
        iProductRepository.createProduct(product);
    }

    @Override
    public void update(Integer id) {
        iProductRepository.updateProduct(id);
    }

    @Override
    public void delete(Integer id) {
        iProductRepository.deleteProduct(id);
    }

    @Override
    public List<Product> search(String name) {
        return iProductRepository.searchProductName(name);
    }


}
